opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1815,6 +1893,25 @@ namespace Opm {
         }
     }
 
+    template<typename TypeTag>
+    void
+    BlackoilWellModel<TypeTag>::
+    addWellPressureEquationsDomain(PressureMatrix& jacobian, const BVector& weights,const bool use_well_weights, int domainIndex) const
+    {
+        int nw =  this->numLocalWellsEnd();
+        int rdofs = local_num_cells_;
+        for ( int i = 0; i < nw; i++ ){
+            int wdof = rdofs + i;
```

Comment:
I am not sure this is correct. The `rdofs` should be the size of the domain, and the `nw` should be the number of wells in the domain. Maybe better to drop this for now, and instead disallow CPRW as linear solver for subdomain nldd solves,  I assume that is the only use case for this function?

Reply:
That is probably correct. I have not yet looked into making CPRW and CPR trueimpes work as local linear solvers yet. This is something that did not work before any of the changes and would result in a crash. This is still the case. 
The changes added in this PR should be a step towards supporting these preconditioners for local linear solvers.
If I have time before this is taken in, I will fix it or else I will do it in a follow up PR.

Reply:
I commented out the code, left some comments about what needs to be fixed, and throw an error if this functions is attempted to be used. I will see if I can implement this in a later PR.